File: runtime/src/main/java/com/siimkinks/sqlitemagic/Expr1.java
package com.siimkinks.sqlitemagic;


import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

final class Expr1 extends Expr {
  @Nullable
  private final String evalArg;

  Expr1(@NonNull Column<?, ?, ?, ?, ?> column, @NonNull String op, @Nullable String evalArg) {
    super(column, op);
    this.evalArg = evalArg;
  }

  @Override
  void addArgs(@NonNull ArrayList<String> args) {
    args.add(evalArg);
  }
}

File: runtime/src/main/kotlin/com/siimkinks/sqlitemagic/Expr1.kt
package com.siimkinks.sqlitemagic

internal class Expr1(
  column: Column<*, *, *, *, *>,
  op: String,
  private val evalArg: String?
) : Expr(
  column = column,
  op = op
) {
  override fun addArgs(args: ArrayList<String?>) {
    args.add(evalArg)
  }
}
